src/views/studio/MyUploadsView/AssetLine/AssetLine.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { formatBytes } from '@/utils/size'
+import {
+  FileLineContainer,
+  FileLinePoint,
+  FileLineLastPoint,
+  FileStatusContainer,
+  FileInfoContainer,
+  FileInfoType,
+  StatusMessage,
+  ProgressbarContainer,
+} from './AssetLine.style'
+import { Icon, Text, CircularProgressbar } from '@/shared/components'
+import { Asset } from '../AssetsGroupUploadBar/AssetsGroupUploadBar'
+
+type AssetLineProps = {
+  isLast?: boolean
+  asset: Asset
+}
+
+const AssetLine: React.FC<AssetLineProps> = ({ isLast = false, asset }) => {
+  const isVideo = asset.type === 'video'
+  const fileInfoIcon = isVideo ? 'video-file' : 'image-file'
+  const fileTypeText = isVideo ? 'Video file' : `${asset.type.charAt(0).toUpperCase() + asset.type.slice(1)} image`
+  const resolution = `${asset.width}x${asset.height}`
+  const size = formatBytes(asset.size)
+
+  let fileStatus
+  let fileStatusMessage = ''
+  switch (asset.status) {
+    case 'uploading':
+      fileStatus = (
+        <ProgressbarContainer>
+          <CircularProgressbar value={asset.progress} />
+        </ProgressbarContainer>
+      )
+      break
+    case 'pending':
+      fileStatus = (
+        <ProgressbarContainer>
+          <CircularProgressbar value={0} />
+        </ProgressbarContainer>
+      )
+      break
+    case 'failed':
+      fileStatus = <Icon name="error-second" />
+      break
+    case 'completed':
+      fileStatus = <Icon name="success" />
+      break
+    case 'reconnecting':
+      fileStatus = <Icon name="error-second" />
+      fileStatusMessage = 'Reconnecting...'
+      break
+    default:
+      fileStatus = <CircularProgressbar value={asset.progress} />
+  }
+
```

Comment:
Don't like these `let`s here. I would do something like this 
```js

  const getStatus = (asset: Asset) => {
    switch (asset.status) {
      case 'uploading':
        return (
          <ProgressbarContainer>
            <CircularProgressbar value={asset.progress} />
          </ProgressbarContainer>
        )
      case 'pending':
        return (
          <ProgressbarContainer>
            <CircularProgressbar value={0} />
          </ProgressbarContainer>
        )
      case 'failed':
        return <Icon name="error-second" />
      case 'completed':
        return <Icon name="success" />
      case 'reconnecting':
        return <Icon name="error-second" />
      default:
        return <CircularProgressbar value={asset.progress} />
    }
  }
// ....

   <FileStatusContainer>{getStatus(asset)}</FileStatusContainer>
```
 and then, where the StatusMessage is 
```js
      {asset.status === 'reconnecting' && <StatusMessage variant="subtitle2">reconnecting</StatusMessage>}
```